### PR TITLE
[3.3.5] Core/Spells: Fix Glyph of Arcane Shot

### PR DESCRIPTION
**Pull request by P-Kito**
Takeover of https://github.com/TrinityCore/TrinityCore/pull/16531 with recommended changes.

Closes https://github.com/TrinityCore/TrinityCore/issues/16500 and https://github.com/TrinityCore/TrinityCore/pull/16531

(Thanks to https://github.com/Eliminationzx)

### CODE REVIEW

File: sql/updates/world/2016_02_15_01_world.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--
+DELETE FROM `spell_proc_event` WHERE `entry`= 56841;
+INSERT INTO `spell_proc_event` VALUES (56841, 0, 9, 2048, 2048, 2048, 256, 0, 0, 0, 0);
+
+DELETE FROM `spell_script_names` WHERE `ScriptName` = 'spell_hun_glyph_of_arcane_shot';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(56841,'spell_hun_glyph_of_arcane_shot');
```
Comment (Vincent-Michael) [inline sql/updates/world/2016_02_15_01_world.sql]: fix end of line

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 2]
```diff
@@ -296,6 +297,69 @@ class spell_hun_disengage : public SpellScriptLoader
...
+            }
+
+            void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+            {
+                PreventDefaultAction();
+                SpellInfo const* procSpell = eventInfo.GetSpellInfo();
+                if (!procSpell)
+                    return;
+
+                if (Unit* target = GetTarget())
```
Comment (Vincent-Michael) [inline src/server/scripts/Spells/spell_hunter.cpp]: gettarget cannot null

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 3]
```diff
@@ -296,6 +297,69 @@ class spell_hun_disengage : public SpellScriptLoader
...
+                bool found = false;
+                Unit::AuraApplicationMap& Auras = eventInfo.GetProcTarget()->GetAppliedAuras();
+                for (Unit::AuraApplicationMap::const_iterator i = Auras.begin(); i != Auras.end(); ++i)
+                {
+                    Aura const* aura = i->second->GetBase();
+                    // Search only Serpent Sting, Viper Sting, Scorpid Sting, Wyvern Sting
+                    flag96 familyFlag = aura->GetSpellInfo()->SpellFamilyFlags;
+                    if (familyFlag.HasFlag(0xC000, 0x1080) && aura->GetSpellInfo()->SpellFamilyName == SPELLFAMILY_HUNTER)
+                    {
+                        found = true;
```
Comment (sirikfoll) [inline src/server/scripts/Spells/spell_hunter.cpp]: You can remove the bool and change to a return true;
And replace return found by return false. Imo it looks cleaner

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 4]
```diff
@@ -296,6 +297,62 @@ class spell_hun_disengage : public SpellScriptLoader
...
+            void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+            {
+                PreventDefaultAction();
+                SpellInfo const* procSpell = eventInfo.GetSpellInfo();
+                if (!procSpell)
+                    return;
+
+                int32 mana = procSpell->CalcPowerCost(GetTarget(), procSpell->GetSchoolMask());
+                int32 basepoint = CalculatePct(mana, aurEff->GetAmount());
+                GetTarget()->CastCustomSpell(GetTarget(), SPELL_HUNTER_GLYPH_OF_ARCANE_SHOT, &basepoint, NULL, NULL, true, NULL, aurEff, GetCasterGUID());
```
Comment (Keader) [inline src/server/scripts/Spells/spell_hunter.cpp]: Some devs often complain about NULL
it is preferable to use nullptr

**Comment by P-Kito** [comment]
Applied your recommendations and squashed commits.

File: sql/updates/world/2016_02_15_01_world.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+--
+DELETE FROM `spell_proc_event` WHERE `entry`= 56841;
+INSERT INTO `spell_proc_event` VALUES (56841, 0, 9, 2048, 2048, 2048, 256, 0, 0, 0, 0);
```
Comment (joschiwald) [inline sql/updates/world/2016_02_15_01_world.sql]: hex looks better, and always add column names to insert querys

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 6]
```diff
@@ -296,6 +297,62 @@ class spell_hun_disengage : public SpellScriptLoader
...
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_HUNTER_GLYPH_OF_ARCANE_SHOT))
+                    return false;
+                return true;
+            }
+
+            bool CheckProc(ProcEventInfo& eventInfo)
+            {
+                Unit::AuraApplicationMap& Auras = eventInfo.GetProcTarget()->GetAppliedAuras();
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_hunter.cpp]: const pls and rename Auras to auras

Comment (joschiwald) [inline src/server/scripts/Spells/spell_hunter.cpp]: and check proctarget against null

File: sql/updates/world/2016_02_15_01_world.sql [hunk 7]
```diff
@@ -0,0 +1,8 @@
+--
```
Comment (Vincent-Michael) [inline sql/updates/world/2016_02_15_01_world.sql]: rename xxxx_world > xxxx_world_335

**Comment by P-Kito** [comment]
All suggestions have been applied, ready for merge. :+1:

**Comment by joschiwald** [comment]
there are some suggestions pending, you know

**Comment by P-Kito** [comment]
Joschiwald is happy now, ready for merge :+1:

**Comment by joschiwald** [comment]
who says?

**Comment by P-Kito** [comment]
Ready for merge

**Pull request merged**